API/Controllers/UserController.cs

```diff
@@ -293,81 +305,82 @@ public async Task<IActionResult> UpdateAccount(int userId, [FromBody] UserResour
             return Ok(mapper.Map<User, UserResourceResult>(user));
         }
 
-        /// <summary>
-        /// This method is responsible for deleting the current account.
-        /// </summary>
-        /// <returns>This method returns status code 200.</returns>
-        /// <response code="200">This endpoint returns status code 200. The current account is deleted.</response>
-        /// <response code="404">The 404 Not Found status code is returned when the current account could not be found.</response>
-        [HttpDelete]
-        [Authorize]
-        [ProducesResponseType((int) HttpStatusCode.OK)]
-        [ProducesResponseType(typeof(ProblemDetails), (int) HttpStatusCode.NotFound)]
-        public async Task<IActionResult> DeleteAccount()
-        {
-            User user = await HttpContext.GetContextUser(userService).ConfigureAwait(false);
-
-            if(await userService.FindAsync(user.Id) == null)
-            {
-                ProblemDetails problem = new ProblemDetails
-                {
-                    Title = "Failed getting the user account.",
-                    Detail = "The database does not contain a user with this user id.",
-                    Instance = "C4C62149-FF9A-4E4C-8C9F-6BBF518BA085"
-                };
-                return NotFound(problem);
-            }
-
-            await userService.RemoveAsync(user.Id);
-            userService.Save();
-            return Ok();
-        }
-       
-
         /// <summary>
         /// This method is responsible for deleting a user account.
         /// </summary>
         /// <returns>This method returns status code 200.</returns>
         /// <response code="200">This endpoint returns status code 200. The account with the specified id is deleted.</response>
         /// <response code="401">The 401 Unauthorized status code is returned when the user is not allowed to delete the account.</response>
         /// <response code="404">The 404 Not Found status code is returned when the user with the specified id could not be found.</response>
+        [HttpDelete]
         [HttpDelete("{userId}")]
         [Authorize]
         [ProducesResponseType((int) HttpStatusCode.OK)]
         [ProducesResponseType(typeof(ProblemDetails), (int) HttpStatusCode.Unauthorized)]
         [ProducesResponseType(typeof(ProblemDetails), (int) HttpStatusCode.NotFound)]
-        public async Task<IActionResult> DeleteAccount(int userId)
+        public async Task<IActionResult> DeleteAccount(int? userId = null)
```

Comment:
Got a status 500 when not passing an userId. "Nullable object must have a value."

Reply:
Just saw someone did use the same GUID in this controller on line: 423, 433. Can you please fix it :)